sdktests/sdk_context_type.go

```diff
@@ -0,0 +1,291 @@
+					in:  fmt.Sprintf(`{"key": "a", "%s": "b"}`, stringAttrName),
+					out: fmt.Sprintf(`{"kind": "user", "key": "a", "%s": "b"}`, stringAttrName),
+				},
+				contextConversionParams{
+					in:  fmt.Sprintf(`{"key": "a", "%s": null}`, stringAttrName),
+					out: `{"kind": "user", "key": "a"}`,
+				})
+		}
+		for _, p := range params {
+			t.Run(p.in, func(t *ldtest.T) {
+				resp := client.ContextConvert(t, servicedef.ContextConvertParams{Input: p.in})
+				require.Equal(t, "", resp.Error)
+				m.In(t).Assert(json.RawMessage(resp.Output), m.JSONEqual(json.RawMessage(p.out)))
+			})
+		}
+	})
+
+	t.Run("invalid context", func(t *ldtest.T) {
+		inputs := []string{
+			``,
+			`{`,    // malformed JSON
+			`true`, // not an object
+
+			// wrong type for built-in property
+			`{"kind": null, "key": "x"}`,
+			`{"kind": true, "key": "x"}`,
+			`{"kind": "org", "key": null}`,
+			`{"kind": "org", "key": 3}`,
+			`{"kind": "org", "name": 3}`,
+			`{"kind": "org", "transient": null}`,
+			`{"kind": "org", "transient": "yes"}`,
+			`{"kind": "org", "_meta": {"secondary": 3}}`,
+			`{"kind": "org", "_meta": {"privateAttributes": 3}}`,
+			`{"kind": "org", "_meta": {"privateAttributes": {}}}`,
+
+			`{"kind": "kind", "key": "x"}`, // kind cannot be "kind"
+			`{"kind": "multi"}`,            // multi-kind with no kinds
+		}
+		for _, input := range inputs {
+			t.Run(input, func(t *ldtest.T) {
+				resp := client.ContextConvert(t, servicedef.ContextConvertParams{Input: input})
+				assert.NotEqual(t, "", resp.Error)
+			})
+		}
+		t.Run("bad characters in kind", func(t *ldtest.T) {
+			// Do this as just one subtest so that the test log isn't incredibly long. We won't cover the
+			// whole Unicode space of course, just enough to verify that non-ASCII and multi-byte characters
+			// are disallowed, as well as ASCII characters that are outside of the allowed set.
+			for ch := 1; ch < 256; ch++ {
+				if (ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z') || (ch >= '0' && ch <= '9') ||
+					ch == '.' || ch == '-' || ch == '_' {
+					continue
+				}
+				kind := "org" + string(rune(ch))
+				// use real JSON encoding for the string in case escaping is needed
+				encoded, _ := json.Marshal(kind)
+
+				input1 := fmt.Sprintf(`{"kind": %s, "key": "x"}`, string(encoded))
+				resp1 := client.ContextConvert(t, servicedef.ContextConvertParams{Input: input1})
+				if resp1.Error == "" {
+					t.Errorf("invalid character was allowed for single-kind (character code %d)", ch)
+				}
+				input2 := fmt.Sprintf(`{"kind": "multi", %s: {"key": "x"}}`, string(encoded))
+				resp2 := client.ContextConvert(t, servicedef.ContextConvertParams{Input: input2})
+				if resp2.Error == "" {
+					t.Errorf("invalid character was allowed for multi-kind (character code %d)", ch)
+				}
+			}
+		})
+	})
+
+	t.Run("invalid old user", func(t *ldtest.T) {
+		inputs := []string{
+			`{}`,
+			`{"key": true}`,
+			`{"key": "a", "anonymous": 3}`,
+			`{"key": "a", "anonymous": 3}`,
```

Comment:
Do you mean `{"key": "a", "anonymous": {}}` in this case?

Reply:
No, it's just an accidentally duplicated line. Thanks.

Reply:
Oh, actually it was supposed to be `secondary`— that was the other built-in one that wasn't covered here.